nginx_config_reloader/__init__.py

```diff
@@ -400,7 +374,11 @@ def process_IN_DELETE(self, event):
     )
 
     if nats_server:
-        start_message_subscribe_loop(nginx_config_changed_handler, nats_server)
+        nats_thread = threading.Thread(
+            target=watch_for_changes,
+            args=(nginx_config_changed_handler, nats_server),
+        )
+        nats_thread.start()
```

Comment:
I don't see this thread being closed anywhere. What if the process is killed, do we have lingering threads?